sdk/python/kfp/dsl/for_loop.py

```diff
@@ -172,26 +185,84 @@ def from_raw_items(
-    ) -> 'LoopArgument':
-        """Creates a LoopArgument object from raw item list."""
+    ) -> 'LoopParameterArgument':
+        """Creates a LoopParameterArgument object from raw item list."""
         if len(raw_items) == 0:
             raise ValueError('Got an empty item list for loop argument.')
 
-        return LoopArgument(
+        return LoopParameterArgument(
             items=raw_items,
             name_code=name_code,
             channel_type=type(raw_items[0]).__name__,
         )
 
-    @classmethod
-    def name_is_loop_argument(cls, name: str) -> bool:
-        """Returns True if the given channel name looks like a loop argument.
 
-        Either it came from a withItems loop item or withParams loop
-        item.
+class LoopArtifactArgument(pipeline_channel.PipelineArtifactChannel):
+    """Represents the artifact arguments that are looped over in a ParallelFor
+    loop.
+
+    The class shouldn't be instantiated by the end user, rather it is
+    created automatically by a ParallelFor ops group.
+
+    To create a LoopArtifactArgument instance, use the factory method::
+
+        LoopArtifactArgument.from_pipeline_channel(...)
+
+
+    Attributes:
+        pipeline_channel: The PipelineArtifactChannel object this
+            LoopArtifactArgument is associated to.
+    """
+
+    def __init__(
+        self,
+        items: pipeline_channel.PipelineArtifactChannel,
+        name_code: Optional[str] = None,
+        name_override: Optional[str] = None,
+        **kwargs,
+    ):
+        """Initializes a LoopArtifactArgument object.
+
+        Args:
+            items: The PipelineArtifactChannel object this LoopArtifactArgument is
+                associated to.
+            name_code: A unique code used to identify these loop arguments.
+                Should match the code for the ParallelFor ops_group which created
+                these LoopArtifactArguments. This prevents parameter name collisions.
+            name_override: The override name for PipelineArtifactChannel.
+            **kwargs: Any other keyword arguments passed down to PipelineArtifactChannel.
         """
-        return  ('-' + cls.LOOP_ITEM_NAME_BASE) in name \
-          or (cls.LOOP_ITEM_PARAM_NAME_BASE + '-') in name
+        if (name_code is None) == (name_override is None):
+            raise ValueError(
+                'Expect one and only one of `name_code` and `name_override` to '
+                'be specified.')
+
+        if name_override is None:
+            super().__init__(name=_make_name(name_code), **kwargs)
+        else:
+            super().__init__(name=name_override, **kwargs)
+
+        self.items_or_pipeline_channel = items
+        self.is_with_items_loop_argument = not isinstance(
+            items, pipeline_channel.PipelineArtifactChannel)
+
+    @classmethod
+    def from_pipeline_channel(
+        cls,
+        channel: pipeline_channel.PipelineArtifactChannel,
+        is_artifact_list: bool = False,
```

Comment:
Is `is_artifact_list` already available on `PipelineArtifactChannel`?

Reply:
Removed the former.

`is_artifact_list` is not available there, but it will always be false because we don't support nested lists yet. So removed that argument and directly pass `is_artifact_list=False` into the subsequent `PipelineArtifactChannel` constructor.